hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirRenameOp.java

```diff
@@ -681,14 +703,46 @@ private static class RenameOperation {
         this.srcIIP = INodesInPath.replace(srcIIP, srcIIP.length() - 1,
             srcChild);
         // get the counts before rename
-        oldSrcCounts.add(withCount.getReferredINode().computeQuotaUsage(bsps));
+        oldSrcCountsInSnapshot.add(withCount.getReferredINode().computeQuotaUsage(bsps));
       } else if (srcChildIsReference) {
         // srcChild is reference but srcChild is not in latest snapshot
         withCount = (INodeReference.WithCount) srcChild.asReference()
             .getReferredINode();
       } else {
         withCount = null;
       }
+      // set quota for src and dst, ignore src is in Snapshot or is Reference
+      this.srcSubTreeCountOp = withCount == null ?
+          quotaPair.getLeft() : Optional.empty();
+      this.dstSubTreeCountOp = quotaPair.getRight();
+    }
+
+    boolean isSameStoragePolicy() {
```

Comment:
Not sure to get what this method purpose. Another side, the first feeling is to calculate if `src` and `dst` has the same StoragePolicy, but it is not actually same implement as `function name`, right?

Reply:
I am confused by the following segment, when src set StoragePolicy then return true directly, which is not same as the name `isSameStoragePolicy` when glance.
```
      if (src.isSetStoragePolicy()) {
        return true;
      }
```

Reply:
This scenario corresponds to the case where the source INode itself has a storage policy. Under such circumstances, we consistently need to use its own storage policy to calculate the quota, rather than the storage policy of the target directory.

Reply:
Got it. Make sense to me.